### PR TITLE
Prep for 2.2.0 Release + Remove jeweler gem 

### PR DESCRIPTION
**Pull request by chen-anders**
`jeweler` doesn't seem to work with Ruby 3+ and it seems like we could get by with just a dynamic gemspec.

I also changed the `sys-proctable` requirements to be at 1.2 minimum due to the `pid` kwarg API change there.

### CODE REVIEW

File: nsq-cluster.gemspec [hunk 1]
```diff
@@ -39,17 +23,14 @@ Gem::Specification.new do |s|
     s.specification_version = 4
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
-      s.add_runtime_dependency(%q<sys-proctable>.freeze, [">= 0"])
-      s.add_development_dependency(%q<jeweler>.freeze, ["~> 2.2"])
+      s.add_runtime_dependency(%q<sys-proctable>.freeze, ["~> 1.2"])
```
Comment (michaelorr) [inline nsq-cluster.gemspec:26]: Should these `sys-proctable` entries be `~> 1.3`?

**Review by michaelorr — APPROVED** [review]

File: nsq-cluster.gemspec (hunk 1 above)
Comment (chen-anders) [reply nsq-cluster.gemspec:26]: The tests also consistently pass for me at 1.2, but I could change it to 1.3 to be consistent with what's in the gemfile.

**Comment by bschwartz** [comment]
Sorry for the late comment, but we should update the CHANGELOG too if we're bumping versions.

**Comment by chen-anders** [comment]
> Sorry for the late comment, but we should update the CHANGELOG too if we're bumping versions.

Yup - I put a PR here for that: https://github.com/wistia/nsq-cluster/pull/21

**Pull request merged**